src/components/shared/thread/ActivityGraphFills.js

```diff
@@ -857,15 +857,16 @@ function _accumulateInBuffer(
     (sampleCategoryStartTime - bufferTimeRangeStart) * xPixelsPerMs;
   let sampleCategoryEndPixel =
     (sampleCategoryEndTime - bufferTimeRangeStart) * xPixelsPerMs;
-  sampleCategoryStartPixel = Math.max(0, sampleCategoryStartPixel);
-  sampleCategoryEndPixel = Math.min(
-    percentageBuffer.length - 1,
-    sampleCategoryEndPixel
-  );
+  if (sampleCategoryStartPixel < 0) {
+    sampleCategoryStartPixel = 0;
+  }
+  if (sampleCategoryEndPixel > percentageBuffer.length - 1) {
+    sampleCategoryEndPixel = percentageBuffer.length - 1;
+  }
   const samplePixel = (sampleTime - bufferTimeRangeStart) * xPixelsPerMs;
-  const intCategoryStartPixel = Math.floor(sampleCategoryStartPixel);
-  const intCategoryEndPixel = Math.floor(sampleCategoryEndPixel);
-  const intSamplePixel = Math.floor(samplePixel);
+  const intCategoryStartPixel = sampleCategoryStartPixel | 0;
+  const intCategoryEndPixel = sampleCategoryEndPixel | 0;
+  const intSamplePixel = samplePixel | 0;
```

Comment:
Yeah there's probably a bug worth filing here... once this lands, I'll push a deploy-preview PR with just this part backed out, so that the issue can be seen more easily.

What I was seeing was that the first 10 or so selection changes were reasonably fast, and then subsequent keypresses spent a lot of time in some GetName fallback path.